Source/OxyPlot/Axes/Axis.cs

```diff
@@ -835,85 +840,13 @@ public bool IsVertical()
         public abstract bool IsXyAxis();
 
         /// <summary>
-        /// Measures the size of the axis (maximum axis label width/height).
+        /// Updates the natural dimensions of axis with respect to graphics context and
+        /// any other graphical housekeeping.
         /// </summary>
         /// <param name="rc">The render context.</param>
-        /// <returns>The size of the axis.</returns>
-        public virtual OxySize Measure(IRenderContext rc)
+        public void Update(IRenderContext rc)
```

Comment:
I think the `DesiredSize` property can be set during the call to `Measure` - I understand this is how it is done in WPF. Revert (remove) the Update method.